src/Microsoft.DotNet.Interactive/Directives/KernelActionDirective.cs

```diff
@@ -81,15 +82,28 @@ private set
 
     public override async Task<IReadOnlyList<CompletionItem>> GetChildCompletionsAsync()
     {
-        var baseCompletions = await base.GetChildCompletionsAsync();
+        List<CompletionItem> completions = [];
+
+        completions.AddRange(await base.GetChildCompletionsAsync());
 
         var subcommandCompletions = Subcommands.Select(s => new CompletionItem(s.Name, WellKnownTags.Method)
         {
             AssociatedSymbol = s,
             Documentation = s.Description
         });
 
-        return subcommandCompletions.Concat(baseCompletions).ToArray();
+        completions.AddRange(subcommandCompletions);
+
+        if (Parent is KernelDirective parent)
```

Comment:
Does order in which the completions are added to the list matter? If yes, should the directive-based ones be added first?

Reply:
It doesn't matter. At this level, the final list hasn't been assembled, so ordering doesn't matter yet, nor does e.g. filtering for partially-typed completions.